### PR TITLE
Fix set character sql parser

### PR DESCRIPTION
**Pull request by soulasuna**
## Fix set character sql parser

Related to #13317.

Changes proposed in this pull request:
-  Set charset sql parser supported for MySQL by `SET {CHARACTER SET | CHARSET} {'charsetName' | DEFAULT};`.
-  Set charset sql parser supported for PostgreSQL by `SET NAMES 'charsetName';`.

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDALStatementSQLVisitor.java [hunk 1]
```diff
@@ -974,13 +974,15 @@ private VariableAssignSegment getVariableAssign(final OptionValueContext ctx) {
     
     @Override
     public ASTNode visitSetCharacter(final SetCharacterContext ctx) {
-        MySQLSetStatement result = new MySQLSetStatement();
+        final MySQLSetStatement result = new MySQLSetStatement();
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDALStatementSQLVisitor.java:977]: @soulasuna Please remove final modifier here.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLDALStatementSQLVisitor.java [hunk 2]
```diff
@@ -77,8 +77,17 @@ public ASTNode visitSet(final SetContext ctx) {
...
+        if (null != ctx.encoding()) {
+            VariableAssignSegment variableAssignSegment = new VariableAssignSegment();
+            VariableSegment variableSegment = new VariableSegment();
+            variableSegment.setVariable("charset");
+            variableAssignSegment.setVariable(variableSegment);
+            String value = ctx.encoding().getText();
+            variableAssignSegment.setAssignValue(value);
+            variableAssigns.add(variableAssignSegment);
+        }
+        result.getVariableAssigns().addAll(variableAssigns);
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLDALStatementSQLVisitor.java:90]: @soulasuna Can you add some sql parse test case for this logic?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDALStatementSQLVisitor.java (hunk 1 above)
Comment (soulasuna) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDALStatementSQLVisitor.java:977]: This final for `<module name="VariableDeclarationUsageDistance"/>` can not removed.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLDALStatementSQLVisitor.java (hunk 2 above)
Comment (soulasuna) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLDALStatementSQLVisitor.java:90]: OK, I will add test.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**